vochain/account_test.go

```diff
@@ -91,6 +94,38 @@ func TestSetAccountInfoTx(t *testing.T) {
 	if acc2 == nil {
 		t.Fatal(ErrAccountNotExist)
 	}
+	if acc2.Balance != 1000 {
+		t.Fatalf("expected balance to be %d got %d", 1000, acc2.Balance)
+	}
+
+	// set account2 as account delegate
+	testSetAccountDelegateTx(t, &signer, app, signer2.Address(), true, acc2.Nonce)
+	// change info to account with account2 as delegate
+	if err := testSetAccountInfoTx(t, &signer2, signer.Address(), nil, app, "changed"); err != nil {
+		t.Fatal(err)
+	}
+	// check delegate can change infoURI
+	acc, err = app.State.GetAccount(signer.Address(), false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if acc.InfoURI != "changed" {
+		t.Fatalf("expected infoURI to be %s got %s", "changed", acc.InfoURI)
+	}
+	// get account2 and check delegate assumed tx costs
+	acc2, err = app.State.GetAccount(signer2.Address(), false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if acc2 == nil {
+		t.Fatal(ErrAccountNotExist)
+	}
+	if acc2.Balance != 900 {
+		t.Fatalf("expected balance to be %d got %d", 900, acc2.Balance)
+	}
```

Comment:
FYI: you can avoid repeating the values with:

```
if got, want := acc2.Balance, 900; got != want {
    t.Fatalf("want balance to be %v, got %v", want, got)
}
```

or with quicktest, `qt.Assert(t, acc2.Balance, qt.Equal, 900)` :)